api/app/GraphQL/Validators/PlaceCandidateValidator.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\GraphQL\Validators;
+
+use App\Enums\PlacementType;
+use App\Enums\PoolCandidateStatus;
+use App\Models\PoolCandidate;
+use Illuminate\Validation\Rule;
+use Nuwave\Lighthouse\Exceptions\ValidationException;
+use Nuwave\Lighthouse\Validation\Validator;
+
+final class PlaceCandidateValidator extends Validator
+{
+    /**
+     * Return the validation rules.
+     *
+     * @return array<string, array<mixed>>
+     */
+    public function rules(): array
+    {
+        $id = $this->arg('id');
+        $candidate = PoolCandidate::find($id);
+        $placedStatuses = array_column(PlacementType::cases(), 'name');
+        $statusesArray = [...$placedStatuses, PoolCandidateStatus::QUALIFIED_AVAILABLE->name];
+
+        if (! (in_array($candidate->pool_candidate_status, $statusesArray))) {
+            throw ValidationException::withMessages(['InvalidStatusForPlacing']);
```

Comment:
Could we use the [`ApiErrorEnums`](https://github.com/GCTC-NTGC/gc-digital-talent/blob/main/api/database/helpers/ApiErrorEnums.php) for this?

Eventually these should be migrated to the new way of doing enums but for now I guess we can just use the existing file.

Reply:
Sure, I don't see why not. Done in https://github.com/GCTC-NTGC/gc-digital-talent/pull/10013/commits/ea36a9c35918dc7a495672f62e041b1986a2c8dd